core/rootchain.go

```diff
@@ -49,11 +49,7 @@ type CurrentBlockEventData struct {
 	hashList [][]byte
 }
 
-func (self *CurrentBlockEventData) DelLastBlockEventHash(currentNum uint64, chainDb store.Database) {
-	if self.blockNum == currentNum {
-		return
-	}
-
+func (self *CurrentBlockEventData) DelLastBlockEventHash(chainDb store.Database) {
 	for _, v := range self.hashList {
 		err := chainDb.Delete(v)
```

Comment:
use database transaction to batch the operations, call `BeginTx()` then `CommitTx()`